src/python/pants/backend/scala/bsp/rules.py

```diff
@@ -154,41 +169,119 @@ def bsp_resolve_field_factory(
 @rule
 async def bsp_resolve_scala_metadata(
     request: ScalaBSPBuildTargetsMetadataRequest,
+    bash: BashBinary,
     jvm: JvmSubsystem,
     scala: ScalaSubsystem,
     build_root: BuildRoot,
 ) -> BSPBuildTargetsMetadataResult:
     resolves = {fs.resolve.normalized_value(jvm) for fs in request.field_sets}
+    jdk_versions = {fs.jdk for fs in request.field_sets}
     if len(resolves) > 1:
         raise ValueError(
             "Cannot provide Scala metadata for multiple resolves. Please set the "
             "`resolve = jvm:$resolve` field in your `[experimental-bsp].groups_config_files` to "
             "select the relevant resolve to use."
         )
-    resolve = list(resolves)[0]
-    scala_version = scala.version_for_resolve(resolve)
+    (resolve,) = resolves
 
+    scala_version = scala.version_for_resolve(resolve)
     scala_runtime = await Get(
         MaterializeScalaRuntimeJarsResult, MaterializeScalaRuntimeJarsRequest(scala_version)
     )
 
+    #
+    # Extract the JDK paths from an lawful-evil process so we can supply it to the IDE
```

Comment:
While neat to say "lawful evil" in a comment, maybe explain why it is lawful evil (i.e., leaking a path to the Coursier cache)?